### PR TITLE
feat: add mongo indexer to mono

### PR DESCRIPTION
**Pull request by bwanglzu**
add mongot indexer to mono repo, adjust latest jina changes, clean up unit tests.

### CODE REVIEW

File: jinahub/indexers/storage/MongoDBStorage/README.md
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/README.md:75]: Fix naming here and in #78

File: jinahub/indexers/storage/MongoDBStorage/manifest.yml
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/manifest.yml:6]: fix urls here

**Review by cristianmtr — CHANGES_REQUESTED** [review]
some changes

File: jinahub/indexers/storage/MongoDBStorage/manifest.yml
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/manifest.yml:6]: this is a wrong URL. 404

File: jinahub/indexers/storage/MongoDBStorage/manifest.yml
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/manifest.yml:8]: this is a wrong URL. 404

File: jinahub/indexers/storage/MongoDBStorage/manifest.yml
Comment (bwanglzu) [reply jinahub/indexers/storage/MongoDBStorage/manifest.yml:6]: after merge it will be available

Comment (cristianmtr) [reply jinahub/indexers/storage/MongoDBStorage/manifest.yml:6]: No, because the folder is `storage/MongoDBStorage`

Comment (bwanglzu) [reply jinahub/indexers/storage/MongoDBStorage/manifest.yml:6]: oh okay good catch

File: jinahub/indexers/storage/MongoDBStorage/README.md
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/README.md:5]:
```suggestion
`mongodb` is a no-sql document data model storage system. It supports JSON and provides an expressive query language for developers.
```

Comment (cristianmtr) [reply jinahub/indexers/storage/MongoDBStorage/README.md:5]: The phrasing was odd

File: jinahub/indexers/storage/MongoDBStorage/README.md
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/README.md:20]: Add this under the heading: 

```
> These are only needed if you download the source code and directly use the class. Not needed if you use the Jina Hub method below.
```

like in the others

File: jinahub/indexers/storage/MongoDBStorage/README.md
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/README.md:31]: Update `Usages` section to be like the other READMEs (only Jinahub usage is shown)

File: jinahub/indexers/storage/MongoDBStorage/README.md
Comment (cristianmtr) [inline jinahub/indexers/storage/MongoDBStorage/README.md:6]: add a quick overview of the class arguments

**Review by cristianmtr — CHANGES_REQUESTED** [review]
some comments

also please mark the comments you've implemented as "addressed"

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Comment by bwanglzu** [comment]
I'll clean up the tests then make it ready to review. SOme of the last weeks updates on Jina core as affected the PR, succh as `hash_content` is not exist anymore, need adjust a little bit.

File: jinahub/indexers/storage/MongoDBStorage/mongohandler.py
Comment (JoanFM) [inline jinahub/indexers/storage/MongoDBStorage/mongohandler.py:44]: no need this double `return`

**Review by JoanFM — DISMISSED** [review]
LGTM! very small comment

**Review by cristianmtr — DISMISSED** [review]
looks good, but don't merge until https://github.com/jina-ai/executors/pull/72 is merged. It contains additional CI checks, which you should also run. So rebase when that is merged.

File: jinahub/indexers/storage/MongoDBStorage/tests/test_mongodb.py
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/tests/test_mongodb.py:15]: Can you add a test for loading from config? For example https://github.com/jina-ai/executors/blob/4c202a563a67b9d07181911488a36b1e23a54f99/jinahub/encoders/text/DPRTextEncoder/tests/unit/test_encoder.py#L25-L30

but use `Path(__file__).parents[1] / 'config.yml'` to be precise about the location of config file

File: jinahub/indexers/storage/MongoDBStorage/Dockerfile
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/Dockerfile:1]:
```suggestion
FROM jinaai/jina:2.0.18 as base
```

File: jinahub/indexers/storage/MongoDBStorage/README.md
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/README.md:17]:
```suggestion
```

File: jinahub/indexers/storage/MongoDBStorage/README.md
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/README.md:25]:
```suggestion
```

**Review by tadejsv — CHANGES_REQUESTED** [review]

File: jinahub/indexers/storage/MongoDBStorage/__init__.py
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/__init__.py:1]: Ah one more thing: can you extract this file to `mongodb_storage.py`, and put in `__init__.py`
```python
from mongodb_storage import  MongoDBStorage
```

File: jinahub/indexers/storage/MongoDBStorage/tests/test_mongodb.py
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/tests/test_mongodb.py:9]: This can then be replaced with 
```suggestion
from ..mongodb_storage import doc_without_embedding
```

**Review by nan-wang — APPROVED** [review]
LGTM👍

File: jinahub/indexers/storage/MongoDBStorage/__init__.py
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/__init__.py:3]: Should be `.monogo_storage`, same below

File: jinahub/indexers/storage/MongoDBStorage/config.yml
Comment (tadejsv) [inline jinahub/indexers/storage/MongoDBStorage/config.yml:4]:
```suggestion
    - __init__.py
```

**Review by tadejsv — APPROVED** [review]
LGTM!

**Pull request merged**
